core/ecosystem/dotnet/localoverlay.go

```
@@ -0,0 +1,172 @@
+	//
+	// The whole tree is scanned rather than the paths being redirected: the file
+	// that shadows the overlay sits above the *consumers*, and a redirect only
+	// says where a package is produced. Looking only along those paths misses
+	// exactly the case worth catching.
+	_ = walkutil.Walk(req.Root, func(path string, d fs.DirEntry) error {
+		if d.IsDir() || filepath.Base(path) != overlayFile {
+			return nil
+		}
+		rel, err := filepath.Rel(req.Root, path)
+		if err != nil || rel == overlayFile {
+			return nil // the overlay itself
```

golangci-lint (2.12.2)

[error] error is not nil (line 76) but it returns nil

(nilerr)